### PR TITLE
[WIP] Tezos development on Windows! 

### PR DESCRIPTION
**Pull request by tcoulter**
See commits! Path issues! Workarounds! Windows! \o/

### CODE REVIEW

**Comment by tcoulter** [comment]
FYI: One big difference to pay attention to between my version and the original: `$PWD` vs `process.env.PWD`. In the former, the working directory was evaluated by the shell during execution of the docker command; in the latter, the working directory is evaluated within the node process before calling out docker. **I'm not confident these are always guaranteed to be the same.** This fear is why Truffle used to (?) pass the working directory everywhere. Is it an irrational fear? Please tell me it is. :)

File: packages/contract/package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   "version": "4.1.5",
   "main": "index.js",
   "scripts": {
-    "compile": "mkdir -p dist && browserify ./index.js -o ./dist/truffle-contract.js && uglifyjs ./dist/truffle-contract.js -o ./dist/truffle-contract.min.js",
+    "compile": "sh -c \"mkdir -p ./dist\" && browserify ./index.js -o ./dist/truffle-contract.js && uglifyjs ./dist/truffle-contract.js -o ./dist/truffle-contract.min.js",
```
Comment (CruzMolina) [inline packages/contract/package.json]: Can you pull out this commit and open a separate PR targeting `develop` in #2780?

File: packages/compile-ligo/index.js [hunk 2]
```diff
@@ -36,7 +36,15 @@ function checkLigo(callback) {
...
-  const command = `docker run -v $PWD:$PWD --rm -i ligolang/ligo:next compile-contract --michelson-format=json ${sourcePath} ${entryPoint}`;
+  // Note that the first volume parameter passed to docker needs to have a path
+  // denoted in the format of of the host filesystem. The latter volume parameter,
+  // as well as the entry point, needs to be denoted in the format of the VM.
+  // Because of this, we rewrite the VM paths relative to a mounted volume called "project".
+  let fullInternalSourcePath =
+    "/project" + sourcePath.replace(process.env.PWD, "");
+  const command = `docker run -v ${
+    process.env.PWD
+  }:/project --rm -i ligolang/ligo:next compile-contract --michelson-format=json ${fullInternalSourcePath} ${entryPoint}`;
```
Comment (CruzMolina) [inline packages/compile-ligo/index.js]: Were you able to smoke test this successfully locally against `tezos-example-box`?

Comment (CruzMolina) [reply packages/compile-ligo/index.js]: Sweet, was able to test the `tezos` job (passed): https://travis-ci.org/trufflesuite/truffle/jobs/640944022?utm_medium=notification&utm_source=github_status

File: packages/contract/package.json (hunk 1 above)
Comment (CruzMolina) [reply packages/contract/package.json]: Nm, got it #2784.

**Comment by CruzMolina** [comment]
Merging since the `@truffle/contract` change has been picked out in #2784.
